File: Codinator/Project/Product Button/Archive/ArchiveViewController.m
//
//  ArchiveViewController.m
//  VWAS-HTML
//
//  Created by Vladimir on 25/05/15.
//  Copyright (c) 2015 Vladimir Danila. All rights reserved.
//

@import Crashlytics;
#import "ArchiveViewController.h"

@interface ArchiveViewController ()
@property (weak, nonatomic) IBOutlet UIButton *closeButton;
@property (weak, nonatomic) IBOutlet UITextView *commitTextView;
@property (weak, nonatomic) IBOutlet UIButton *commitButton;
@end


@implementation ArchiveViewController
@synthesize projectManager,commitTextView,closeButton,commitButton;


- (void)viewDidLoad {
    [super viewDidLoad];

    closeButton.layer.cornerRadius = 5;
    
    commitTextView.layer.cornerRadius = 10;
    commitTextView.layer.masksToBounds = YES;
    
    commitTextView.delegate = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)archiveDidPush:(id)sender {
    
    [commitTextView resignFirstResponder];
    
    [projectManager archiveWorkingCopyWithCommitMessge:commitTextView.text];
    
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil message:@"Project is 📦 now." preferredStyle:UIAlertControllerStyleAlert];
                UIAlertAction *closeAlert = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleCancel handler:^(UIAlertAction * __nonnull action) {
                    [self dismissViewControllerAnimated:true completion:^{
                        
                        [Answers logCustomEventWithName:@"Archived Version"
                                       customAttributes:@{
                                                          @"Commit text" : commitTextView.text}];
                        
                        [self dismissViewControllerAnimated:true completion:nil];
                    }];
                }];
    
    
    
    [alert addAction:closeAlert];
    [self presentViewController:alert animated:YES completion:nil];
        
}




- (void)textViewDidBeginEditing:(UITextView *)textView{
    textView.text = @"";
}



- (IBAction)closeButtonDidPush:(id)sender {
    [self dismissViewControllerAnimated:true completion:nil];
}




@end

File: Codinator/Project/Product Button/Archive/ArchiveViewController.swift
//
//  ArchiveViewController.swift
//  Codinator
//
//  Created by Vladimir Danila on 20/04/16.
//  Copyright © 2016 Vladimir Danila. All rights reserved.
//

import UIKit

class ArchiveViewController: UIViewController {

    var projectManager: Polaris!
    
    @IBOutlet var textView: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        textView.layer.masksToBounds = true
        textView.layer.cornerRadius = 5
    
    }

    
    @IBAction func cancelDidPush(sender: AnyObject) {
        self.dismissViewControllerAnimated(true, completion: nil)
    }
    
    
    @IBAction func archiveDidPush(sender: UIBarButtonItem) {
        textView.resignFirstResponder()
        
        if textView.text.isEmpty {
            Notifications.sharedInstance.alertWithMessage("Commit message can't be emtpy!", title: "", viewController: self)
        }
        else {
            projectManager.archiveWorkingCopyWithCommitMessge(textView.text)
                self.dismissViewControllerAnimated(true, completion: nil)
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
}
